dvc/stage/cache.py

```
@@ -177,11 +177,15 @@ def _transfer(func, from_remote, to_remote):
 
     def push(self, remote):
         remote = self.repo.cloud.get_remote(remote)
-        return self._transfer(remote.upload, self.repo.cache.local, remote)
+        return self._transfer(
+            remote.tree.upload, self.repo.cache.local.tree, remote.tree
+        )
```

Just changed it to `.tree` instead of changing everywhere in `_transfer()`.

---

@skshetry `tree.upload/download` are not meant to swallow errors, unlike the remote, so it won't work :slightly_frowning_face:

---

@efiop, so, what do you propose? I'm still wrapping my heads around tree. Is there any other interface, or are you proposing to accumulate the errors and continue and show error later?

---

@skshetry We just need to wrap those with `_log_exceptions` helper from `local.py`. That will do the trick for now.